modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -161,6 +182,89 @@ public int call(final String[] args, final Cfg cfg) {
                         }
                     }
                     break;
+
+                // Determines network config folder path
+                case "-n":
+                case "--network":
+                    if ( (args.length == 2 && isValid(args[1])) || (args.length == 4 && isValid(args[1]) && isValid(args[3])) ) {
+
+                        switch (args[1].toLowerCase()) {
+                            case "mainnet":
+                            case "conquest":
+
+                                // Recursively call to change database path
+                                if (args.length == 4 && (args[2].equals("--datadir") || args[2].equals("-d"))) {
+                                    // Modify config path to inside datadir folder
+                                    CfgAion.setNetwork(args[1]);
+                                    CfgAion.setConfFilePath(BASE_PATH + "/" + args[3] + "/config/" + args[1] + "/config.xml");
+                                    CfgAion.setGenesisFilePath(BASE_PATH + "/" + args[3] + "/config/" + args[1] + "/genesis.json");
+                                    Keystore.setKeystorePath(BASE_PATH + "/" + args[3] + "/keystore");
+                                    String[] newArgs = Arrays.copyOfRange(args, 2, args.length);
+                                    call(newArgs, cfg);
+                                } else {
+                                    // Change to specified network path
+                                    CfgAion.setNetwork(args[1]);
+                                    CfgAion.setConfFilePath(path + "/config/" + args[1] + "/config.xml");
+                                    CfgAion.setGenesisFilePath((path + "/config/" + args[1] + "/genesis.json"));
+                                }
+                                break;
+                            default:
+                                System.out.println("Invalid network '" + args[1] + "': Set to default network");
+                                break;
+                        }
+                    } else {
+
+                        // Set to default database && log && network
+                        System.out.println("Invalid arguments: Set to default network");
+                        cfg.getDb().setDatabasePath("database");
+                        cfg.getLog().setLogPath("log");
+                        cfg.toXML(null);
+                        return 1;
+                    }
+                    break;
+
+                // Determines database folder path
+                case "-d":
+                case "--datadir":
```

Comment:
Same here, if invalid args are given the kernel still is started up

Reply:
the kernel is defaulted to mainnet + database when there is an invalid argument; i think this is the behavior initially wanted but i will double check and see if i should change this